consensus/consortium/v2/consortium.go

```diff
@@ -1461,3 +1595,60 @@ func encodeSigHeader(w io.Writer, header *types.Header, chainId *big.Int) {
 		panic("can't encode: " + err.Error())
 	}
 }
+
+// IsPeriodBlock returns indicator whether a block is a period checkpoint block or not,
+// which is the first checkpoint block (block % EpochV2 == 0) after 00:00 UTC everyday.
+func (c *Consortium) IsPeriodBlock(chain consensus.ChainHeaderReader, header *types.Header) bool {
+	number := header.Number.Uint64()
+	if number%c.config.EpochV2 != 0 || !chain.Config().IsTripp(header.Number) {
+		return false
+	}
+	snap, err := c.snapshot(chain, number-1, header.ParentHash, nil)
+
+	// If error happens when derive snapshot or current period is absent, we recursively find
+	// the nearest epoch block; and determine whether the header is one day ahead of that neighbor.
+	if err != nil {
+		log.Warn("Fail to get snapshot at", "block", number-1, "err", err)
+		var (
+			parentNumber uint64        = number - 1
+			parentHash   common.Hash   = header.ParentHash
+			parent       *types.Header = chain.GetHeader(parentHash, parentNumber)
+		)
+		for parentNumber%c.config.EpochV2 != 0 && parent != nil {
+			parentNumber = parent.Number.Uint64() - 1
+			parentHash = parent.ParentHash
+			parent = chain.GetHeader(parentHash, parentNumber)
+		}
+		return uint64(header.Time/dateInSeconds) > uint64(parent.Time/dateInSeconds)
```

Comment:
The above loop can break if parent == nil. In that case, parent.Time results into a nil pointer dereference.

Reply:
fixed